### PR TITLE
[test] #137 implement newVoteReceived event on Fund contract

### LINKED ISSUE #137: [feat] Emit event when fund receives a vote
### This issue will be part of ODHack9.0, please apply via Onlydust app
- Emit an event when a fund contract receives a new vote in the ```receiveVote``` method.
- The event should contain: voter(caller address), the fund contract address that receives the vote, current votes.
- Please read [contributors guide](https://github.com/web3wagers/gostarkme/blob/dev/CONTRIBUTORS_GUIDE.md) before asking for an issue

### Acceptance Criteria
- Code builds successfully.
- Event struct has all the attributes needed.

### PR DESCRIPTION
**Pull request by jorgezerpa**
# Pull Request

- [ ] Closes #137 
- [ ] Added tests (if necessary) -> none
- [ ] Run tests -> none
- [ ] Run formatting -> none
- [ ] Commented the code -> none

## Changes description
* create an event NewVoteReceived in the Fund contract
* Emit such event when call receiveVote function

## Current output
image
image

## Time spent breakdown

1 hour

## Comments

I'm happy to contribute! I stay attentive to any feedback or needed modification

### CODE REVIEW

File: contracts/src/fund.cairo [hunk 1]
```diff
@@ -68,6 +68,21 @@ mod Fund {
...
+    // *************************************************************************
+    #[event]
+    #[derive(Drop, starknet::Event)]
+    enum Event {
+        NewVoteReceived: NewVoteReceived
+    }
+
+    #[derive(Drop, starknet::Event)]
+    pub struct NewVoteReceived {
+        pub voter: ContractAddress,
```
Comment (EmmanuelAR) [inline contracts/src/fund.cairo:82]: Excellent work @jorgezerpa!
For this event we what this values in it -> The event should contain: voter(caller address), the fund contract address that receives the vote, current votes.
please add the pending ones

**Review by EmmanuelAR — CHANGES_REQUESTED** [review]

**Comment by EmmanuelAR** [comment]
Hello @jorgezerpa thanks for the pr, so fast, there are some changes request. Thks again!

File: contracts/src/fund.cairo (hunk 1 above)
Comment (EmmanuelAR) [reply contracts/src/fund.cairo:82]: Also the voter have to be the key of the event, please add #[key] in the attribute, you can see some examples

**Review by EmmanuelAR — CHANGES_REQUESTED** [review]

**Comment by jorgezerpa** [comment]
Hello @EmmanuelAR I got it!🫡 I'll try to get it done by tomorrow before half of the day🚀

**Comment by EmmanuelAR** [comment]
Take your time Sr!

**Comment by jorgezerpa** [comment]
GM! changes ready:
image

Just 2 things before make the push:
1. When I access to "self.up_votes.read()" should I add 1 on the event? (like "self.up_votes.read() + 1")  or with the line "self.up_votes.write(self.up_votes.read() + 1);" that is above in the same function the value is already added when I access try to again?
2. Check the name of the keys, let me know if are OK or maybe you would like to change someone

**Comment by EmmanuelAR** [comment]
Hey @jorgezerpa 
1. For the votes just call the votes.read() because we already increment it.
2. I cant check the name of the keys because you have to push the code
3. please run scarb fmt

**Comment by jorgezerpa** [comment]
@EmmanuelAR pushed!🙌

**Review by EmmanuelAR — APPROVED** [review]
Exc work @jorgezerpa LGTM!

**Comment by EmmanuelAR** [comment]
hey @jorgezerpa please merge your local branch with the latest version of the code, we add some events so the event trait is already create and you are adding twice fix like this example:

Captura de pantalla 2024-10-25 a la(s) 10 16 22 a  m

Also run scarb build and fmt

**Comment by jorgezerpa** [comment]
@EmmanuelAR Merged and pushed🫡

**Pull request merged**